dataland-frontend/tests/e2e/specs/high/Prepopulation.ts

```diff
@@ -12,56 +19,91 @@ describe('Population Test', () => {
+        for (let i = 0; i < dataArray.length; i += chunkSize) {
+            const chunk = dataArray.slice(i, i + chunkSize);
+            await Promise.all(chunk.map(async (element:object) => {
+                    await fetch(`${Cypress.env("API")}/${endpoint}`, {
+                        method: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        },
+                        body: JSON.stringify(element)
+                    }).then(response => {
+                        assert(response.status.toString() === "200" )
+                    })
+                })
+            )
+        }
+        const millis = Date.now() - start
+        console.log(`seconds elapsed = ${Math.floor(millis / 1000)}`)
+    }
 
-    it('Populate Companies', function (){
-        for (const index in companiesData) {
-            cy.request('POST', `${Cypress.env("API")}/companies`, companiesData[index]).its('status').should("equal", 200)
 
-        }
-        console.log(companiesData)
+    it('Populate Companies',  async() => {
+        await uploadData(companiesData, "companies")
     });
 
-    it('Populate EU Taxonomy Data', function (){
-        for (const index in eutaxonomiesData) {
-            cy.request('POST', `${Cypress.env("API")}/data/eutaxonomies`, eutaxonomiesData[index]).its('status').should("equal", 200)
-        }
-        console.log(eutaxonomiesData)
+
+    it('Retrieve company ID list', () => {
+        cy.request('GET', `${Cypress.env("API")}/companies`).then((response) => {
+            companyIdList = response.body.map((e: any, index:number) => {
+                if (typeof eutaxonomiesData[index] == "object"){
+                    eutaxonomiesData[index].companyId = e.companyId
+                }
+                return e.companyId
+            })
+        })
+    });
+
+    it('Populate EU Taxonomy Data',  async() => {
+        await uploadData(eutaxonomiesData, "data/eutaxonomies")
     });
 
     it('Retrieve data ID list', () => {
         cy.request('GET', `${Cypress.env("API")}/metadata`).then((response) => {
-            idList = response.body.map(function (e:string){
-                return parseInt(Object.values(e)[0])
+            dataIdList = response.body.map(function (e: any) {
+                return e.dataId
             })
         })
     });
 
-
 });
 
 describe('EU Taxonomy Data', () => {
     it('Check Data Presence and Link route', () => {
-        cy.visit("/data/eutaxonomies/"+idList[0])
-        cy.get('h3').contains("Revenue")
-        cy.get('h3').contains("CapEx")
-        cy.get('h3').contains("OpEx")
-        cy.get('.d-card').should('contain', 'Eligible')
-        cy.get('.d-card .p-progressbar').should('exist')
+        cy.intercept('**/api/data/eutaxonomies/*').as('retrieveData')
+        cy.visit("/data/eutaxonomies/"+dataIdList[0])
+        cy.wait(1000)
+        cy.wait('@retrieveData', {timeout: 2000}).then(() => {
```

Comment:
wir warten immer 1s, und aber dann nochmal bis zu 2s bis der request zurück ist? Hä?

Reply:
die statischen waits sind mit dem Zweck die Page laden zu lassen (wie andere Kommentar). Z.B braucht hier cypress bisschen Zeit eutaxonomy Seite zu laden. wir wollten das statische wait vermeiden deswegen haben wir gedacht das würde reichen wenn man auf den request wartet. Aber das dat hat nicht gereicht, weil das brauchte die Seite zu laden. ich mache wait auf den Request weg. Aber man später eine schönere Lösung für die statischen waits finden

Reply:
Können wir denn nicht einfach darauf warten, was passieren soll?
cy.get('[data-test=submitIsVisible]', { timeout: 10000 }).should('be.visible');

Reply:
ja - so ists super :-)